pkg/util/provider/machineutils/utils.go

```diff
@@ -69,6 +70,9 @@ const (
 
 	// MachineLabelKey defines the labels which contains the name of the machine of a node
 	MachineLabelKey = "node.gardener.cloud/machine-name"
+
+	// LabelKeyMachineSetScaleUpDisabled is the label key that indicates scaling up of the machine set is disabled.
+	LabelKeyMachineSetScaleUpDisabled = "node.machine.sapcloud.io/scale-up-disabled"
```

Comment:
Can you please let me know the purpose you intend to use this for?

Reply:
This is needed so that when we move the machine from oldMachineSet to newMachineSet, during the period of the machine being moved to down scaling of the oldMachineSet, new machine should not be created for the oldMachineSet for the difference of the count of the machines it has and the number of replicas in its spec.